### PR TITLE
Enable students to edit pages

### PR DESCRIPTION
**Pull request by shichongrui**
refs: MBL-12293
affects: Student
release note: Made it possible for students to edit pages
test plan: see ticket

### CODE REVIEW

**Comment by inst-danger** [comment]
#### Release Note: 
Made it possible for students to edit pages

#### Affected Apps: Student

[MBL-12293](https://instructure.atlassian.net/browse/MBL-12293)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 93.6% | 93.62% | -0.01%
**Student** | 92.2% | 92.2% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.65% | 92.63% | 0.02%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: rn/Teacher/src/modules/pages/details/PageDetails.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ export class PageDetails extends Component<Props> {
   constructor (props: Props) {
     super(props)
 
-    if (isStudent) {
+    if (isStudent()) {
```
Comment (armstrongnate) [inline rn/Teacher/src/modules/pages/details/PageDetails.js]: Oh boy. That's awesome.

File: rn/Teacher/src/modules/pages/edit/PageEdit.js [hunk 2]
```diff
@@ -142,51 +147,54 @@ export class PageEdit extends Component<Props, State> {
...
-                onValueChange={this.handleEditingRolesChange}
-                testID='pages.edit.editing_roles.picker'
-              >
-                {Object.keys(editingRoles()).map(key => (
-                  <PickerItem
-                    key={key}
-                    value={key}
-                    label={editingRoles()[key]}
+            {isTeacher() &&
+              <React.Fragment>
```
Comment (armstrongnate) [inline rn/Teacher/src/modules/pages/edit/PageEdit.js]: Interesting. What is `React.Fragment`?

File: rn/Teacher/src/modules/inbox/__tests__/__snapshots__/Compose.test.js.snap [hunk 3]
```diff
@@ -284,15 +284,96 @@ exports[`Compose renders after picking a course 1`] = `
...
                 },
-                undefined,
+                Object {
+                  "color": "#2D3B45",
+                },
               ]
             }
           >
-            Select a Course
+            Learn React Native
```
Comment (armstrongnate) [inline rn/Teacher/src/modules/inbox/__tests__/__snapshots__/Compose.test.js.snap]: The changes in this file seem odd. Do you know what they are about?

File: rn/Teacher/src/modules/pages/edit/PageEdit.js (hunk 2 above)
Comment (shichongrui) [reply rn/Teacher/src/modules/pages/edit/PageEdit.js]: `React.Fragment` is a component that essentially just wraps other components. So in this case when we want to conditionally render a couple of views that are all siblings, without `React.Fragment` we have 3 options.
 - We conditionally render each sibling individually which isn't great. 
 - We wrap them in a `View` and conditionally render that, but then we have an extra `View` hanging around that isn't really needed
 - We conditionally render an array of views. But with this then you have to provide a key for each element which isn't great.

So instead we use `React.Fragment` which is a wrapping component that renders nothing. Gives us the ability to conditionally render a bunch of sibling components without the extra `View`.

File: rn/Teacher/src/modules/inbox/__tests__/__snapshots__/Compose.test.js.snap (hunk 3 above)
Comment (shichongrui) [reply rn/Teacher/src/modules/inbox/__tests__/__snapshots__/Compose.test.js.snap]: These changes happened after I updated enzyme to work with `React.Fragment`. I think these changes are actually correct and that what was there before was not. The test is testing something after choosing a course, but the original snapshot still said `Select a Course`. So it seemed like the enzyme update actually made the test work the way it was intended.

**Pull request merged**